tests/Microsoft.Identity.Test.Unit.net45/AppConfigTests/PublicClientApplicationBuilderTests.cs

```diff
@@ -0,0 +1,171 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using System.Net.Http;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.AppConfig;
+using Microsoft.Identity.Client.Core;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Unit.AppConfigTests
+{
+    [TestClass]
+    [TestCategory("BuilderTests")]
+    public class PublicClientApplicationBuilderTests
+    {
+        [TestMethod]
+        public void TestConstructor()
+        {
+            var pca = PublicClientApplicationBuilder.Create(MsalTestConstants.ClientId).Build();
+            Assert.AreEqual(MsalTestConstants.ClientId, pca.ClientId);
+            Assert.IsNotNull(pca.UserTokenCache);
+
+            // Validate Defaults
+            Assert.AreEqual(LogLevel.Warning, pca.AppConfig.LogLevel);
+            Assert.IsNull(pca.AppConfig.ClientCredential);
+            Assert.AreEqual(MsalTestConstants.ClientId, pca.AppConfig.ClientId);
+            Assert.IsNull(pca.AppConfig.Component);
+            Assert.AreEqual(false, pca.AppConfig.EnablePiiLogging);
+            Assert.IsNull(pca.AppConfig.HttpClientFactory);
+            Assert.AreEqual(false, pca.AppConfig.IsDefaultPlatformLoggingEnabled);
+            Assert.IsNull(pca.AppConfig.LoggingCallback);
+            Assert.AreEqual(Constants.DefaultRedirectUri, pca.AppConfig.RedirectUri);
+            Assert.IsNull(pca.AppConfig.TelemetryCallback);
+            Assert.AreEqual(null, pca.AppConfig.TenantId);
+        }
+
+        [TestMethod]
+        public void TestWithDifferentClientId()
+        {
+            var pca = PublicClientApplicationBuilder.Create("this is a test client id").Build();
+            Assert.AreEqual("this is a test client id", pca.ClientId);
+        }
+
+        [TestMethod]
+        public void TestConstructor_ClientIdOverride()
+        {
+            var pca = PublicClientApplicationBuilder.Create(MsalTestConstants.ClientId).WithClientId("some other client id").Build();
+            Assert.AreEqual("some other client id", pca.ClientId);
+        }
+
+        [TestMethod]
+        public void TestConstructor_WithComponent()
+        {
+            var pca =
+                PublicClientApplicationBuilder.Create(MsalTestConstants.ClientId).WithComponent("my component name").Build();
+            Assert.AreEqual("my component name", pca.AppConfig.Component);
+        }
+
+        [TestMethod]
+        public void TestConstructor_WithDebugLoggingCallback()
+        {
+            var pca = PublicClientApplicationBuilder.Create(MsalTestConstants.ClientId).WithDebugLoggingCallback().Build();
+            Assert.IsNotNull(pca.AppConfig.LoggingCallback);
```

Comment:
Where will the developer hook up the logger? Are there any tests validating that attaching a default logger generates something on the logger?

Reply:
This is for the debug logger (hooks up a default logger that calls Debug.WriteLine()).  There are other tests that implicitly test the logger callback (and telemetry callback) work because they mock the logger callback and look for messages.